js/apps/system/_admin/aardvark/APP/react/src/views/collections/addCollection/AddCollectionModal.tsx

```diff
@@ -0,0 +1,155 @@
+import { Button, Flex, Heading, Stack } from "@chakra-ui/react";
+import { CreateCollectionOptions } from "arangojs/collection";
+import { Form, Formik } from "formik";
+import React from "react";
+import { mutate } from "swr";
+import * as Yup from "yup";
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader
+} from "../../../components/modal";
+import { getCurrentDB } from "../../../utils/arangoClient";
+import { AddCollectionForm } from "./AddCollectionForm";
+
+const INITIAL_VALUES = {
+  name: "",
+  type: "document" as "document" | "edge",
+  waitForSync: false,
+  numberOfShards: 1,
+  shardKeys: ["_key"],
+  replicationFactor: window.frontendConfig.defaultReplicationFactor,
+  writeConcern: 1,
+  distributeShardsLike: "",
+  isSatellite: false,
+  smartJoinAttribute: ""
+} satisfies Partial<CreateCollectionOptions> & Record<string, any>;
+
+export const AddCollectionModal = ({
+  isOpen,
+  onClose
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+}) => {
+  const handleSubmit = async (values: typeof INITIAL_VALUES) => {
+    const currentDB = getCurrentDB();
+    const {
+      name,
+      type,
+      waitForSync,
+      distributeShardsLike,
+      numberOfShards,
+      replicationFactor,
+      writeConcern,
+      isSatellite,
+      ...extra
+    } = values;
+    const clusterOptions: Partial<CreateCollectionOptions> = window.App
+      .isCluster
+      ? distributeShardsLike
+        ? { distributeShardsLike, ...extra }
+        : {
+            numberOfShards,
+            replicationFactor: (isSatellite
+              ? "satellite"
+              : replicationFactor) as any, // TODO workaround for arangojs 7
+            writeConcern,
+            ...extra
+          }
+      : {};
+    try {
+      await currentDB.collection(name).create({
+        type: type === "edge" ? 3 : 2,
```

Comment:
can we use something from arangojs instead of using the number `3`, `2`. For some cases it has an exposed `collectionType.EDGE` kindof things, if I'm not wrong